pkg/controller/syncmachineset/syncmachineset_controller.go

```diff
@@ -0,0 +1,555 @@
+	rCluster, err := remoteClusterAPIClient.ClusterV1alpha1().Clusters(remoteClusterAPINamespace).Get(coCluster.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		// Cluster does not exist, create it:
+		msLog.Infof("creating new cluster API object in remote cluster")
+		rCluster = buildClusterAPICluster(coCluster)
+		_, err := remoteClusterAPIClient.ClusterV1alpha1().Clusters(remoteClusterAPINamespace).Create(rCluster)
+		if err != nil {
+			return fmt.Errorf("error creating cluster object in remote cluster: %v", err)
+		}
+	} else if err != nil {
+		return err
+	} else {
+		// TODO: patch/update here?
+		msLog.Debugf("cluster API object already exists in remote cluster")
+	}
+
+	return nil
+}
+
+// Creates machinesets in the remote.
+func (c *Controller) syncComputeMachineSet(ms *cov1.MachineSet) error {
+	msLog := logging.WithMachineSet(c.logger, ms)
+	msLog.Debug("loaded generic machineset")
+
+	// Lookup the CO cluster object for this machine set and use
+	// that to get the master machine set for the cluster. Continue
+	// syncing generic machine set if the master machine set is
+	// installed (ms.Status.ClusterAPIInstalled == true).
+	// TODO: Set a flag on cluster object once it has been synced
+	// so that we don't have to retrieve the master machine set to
+	// know if we can sync the other machine sets.
+	coCluster, err := c.client.ClusteroperatorV1alpha1().Clusters(ms.Namespace).Get(ms.Labels[controller.ClusterNameLabel], metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error looking up machineset's cluster: %v", err)
+	}
+	coMasterMachineSet, err := c.client.ClusteroperatorV1alpha1().MachineSets(coCluster.Namespace).Get(coCluster.Status.MasterMachineSetName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error looking up cluster's master machineset: %v", err)
+	}
+
+	// handleErr will requeue machine sets after waiting when the
+	// remote cluster API is not installed.
+	if !coMasterMachineSet.Status.ClusterAPIInstalled {
+		return NewErrClusterAPINotInstalled(errorMsgClusterAPINotInstalled)
+	}
+
+	remoteClusterAPIClient, remoteDeploymentClient, err := c.buildClients(ms)
+	if err != nil {
+		return err
+	}
+
+	// handleErr will requeue machine sets after waiting when the
+	// remote cluster API deployment is not ready.
+	remoteClusterAPIDeployment, err := remoteDeploymentClient.AppsV1().Deployments(remoteClusterAPINamespace).Get(remoteClusterAPIDeploymentName, metav1.GetOptions{})
+	if err != nil {
+		return NewErrClusterAPINotInstalled(fmt.Sprintf("unable to retrieve remote cluster API deployment: %v", err))
+	}
+	if remoteClusterAPIDeployment.Status.ReadyReplicas < 1 {
+		return NewErrClusterAPINotInstalled(errorMsgClusterAPINotInstalled)
+	}
+
+	// Create machineSet object in remote if the machineSet does not already exist.
+	remoteMachineSet, err := remoteClusterAPIClient.ClusterV1alpha1().MachineSets(remoteClusterAPINamespace).Get(ms.Name, metav1.GetOptions{})
```

Comment:
Same concern here as with master machine set. We should be deliberate about verifying that the machine set in the remote cluster is up-to-date on a schedule rather than on every controller sync of the compute machine set.